src/expression/expression_tree.rs

```diff
@@ -173,12 +173,20 @@ impl fmt::Display for Atom {
     }
 }
 
+type AlVec<T> = alloc::vec::Vec<T>;
```

Comment:
I think we should have internal consistency on the vec type used inside Expression

Reply:
That does make sense. What would be the recommended default size of the `heapless::Vec` then?

Reply:
actually, you can keep it as-is. this is something I need to revise later.

Reply:
Ok, so are we going to be utilizing `alloc::vec::Vec` instead of `heapless::Vec` now? I remember the initial plan was to get rid of the `alloc` dependency all together but we very quickly backtracked on that plan.

Reply:
I'm not completely sure. I'll add your current implementation, and I'll see what I can do about it later.